src/pytest_mock_resources/fixture/database/relational/generic.py

```diff
@@ -154,6 +155,61 @@ def manage(self, session=None):
         finally:
             self.engine.dispose()
 
+    def manage_sync(self, session=None):
+        try:
+            self._run_actions()
+
+            if session:
+                if isinstance(session, sessionmaker):
+                    session_factory = session
+                else:
+                    session_factory = sessionmaker(bind=self.engine)
+
+                Session = scoped_session(session_factory)
+                session = Session(bind=self.engine)
+                yield session
+                session.close()
+            else:
+                yield self.engine
+        finally:
+            self.engine.dispose()
+
+    async def manage_async(self, session=None):
+        try:
+            self._run_actions()
+
+            async_engine = self._get_async_engine()
+
+            if session:
+                if isinstance(session, sessionmaker):
+                    session_factory = session
+                else:
+                    session_factory = sessionmaker(
+                        async_engine, expire_on_commit=False, class_=AsyncSession
+                    )
+                async with session_factory() as session:
+                    yield session
+            else:
+                yield async_engine
+        finally:
+            self.engine.dispose()
+
+    def _get_async_engine(self, isolation_level=None):
+        URI_TEMPLATE = (
+            "postgresql+asyncpg://{username}:{password}@{host}:{port}/{database}?ssl=disable"
```

Comment:
we **should** be doing this elsewhere, but there's no reason to generate our own URI string ourselves (technically you should be url-encoding the password!), you can `from sqlalchemy.engine.url import URL`, send these params into `URL(...)`, and give that directly to `create_[async_]engine`